packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/calendar.tests.js

```diff
@@ -1383,6 +1383,35 @@ QUnit.module('Preserve time component on value change', {
 
         assert.deepEqual(calendar.option('value'), new Date(2015, 8, 2, 12, 57));
     });
+
+    QUnit.test('should not switch to the earliest date after setting same value programmatically (T1279950)', function(assert) {
+        this.calendar.option({
+            selectionMode: 'multiple',
+            value: [new Date(2025, 1, 10), new Date(2025, 2, 10)],
+        });
+
+        const calendar = this.calendar;
+        const $nextMonthButton = this.$element.find(toSelector(CALENDAR_NAVIGATOR_NEXT_MONTH_CLASS));
+
+        assert.ok(
+            dateUtils.sameMonth(calendar.option('currentDate'), new Date(2025, 1, 10)),
+            'initially navigated to the earliest date'
+        );
+
+        $($nextMonthButton).trigger('dxclick');
+
+        assert.ok(
+            dateUtils.sameMonth(calendar.option('currentDate'), new Date(2025, 2, 10)),
+            'navigated to the next month'
+        );
+
+        calendar.option('value', [new Date(2025, 1, 10), new Date(2025, 2, 10)]);
+
+        assert.ok(
+            dateUtils.sameMonth(calendar.option('currentDate'), new Date(2025, 2, 10)),
+            'did not navigate back to the earliest date'
+        );
+    });
```

Comment:
[test coverage]
Please add additionally a test for view navigation after deselect